pkg/manager/interfaces/execution.go

```diff
@@ -13,6 +13,8 @@ type ExecutionInterface interface {
 		*admin.ExecutionCreateResponse, error)
 	RelaunchExecution(ctx context.Context, request admin.ExecutionRelaunchRequest, requestedAt time.Time) (
 		*admin.ExecutionCreateResponse, error)
+	RecoverExecution(ctx context.Context, request admin.ExecutionRecoverRequest, requestedAt time.Time) (
```

Comment:
I think we should add extensive documentation at the interface level to explain what each method does. Lets start with recover

Reply:
done